### PR TITLE
chore: Subdomain blacklist when creating org

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

When creating an organization, we shouldn't let users pick reserved subdomains such as docs, design, etc. This PR introduces a blacklist of subdomains to check before creating an org.

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

Any user shouldn't be able to pick reserved subdomains as valid organization when creating one.

## Mandatory Tasks
- [X] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/lib/constants.ts [hunk 1]
```diff
@@ -70,3 +70,33 @@ export const IS_STRIPE_ENABLED = !!(
...
+  "go",
+  "status",
+  "api",
+  "saml",
+  "www",
+  "matrix",
+  "developer",
+  "cal",
+  "my",
+  "team",
```
Comment (baileypumfleet) [inline packages/lib/constants.ts:101]:
```suggestion
  "team",
  "support",
  "security",
  "blog",
  "learn",
  "admin",
```

**Review by pumfleet — COMMENTED** [review]
Nice! Just added some more suggestions

File: packages/lib/constants.ts (hunk 1 above)
Comment (leog) [reply packages/lib/constants.ts:101]: Added, feel free to approve.

**Review by zomars — COMMENTED** [review]
Should selfhosters modify this file if they don't want to blacklist?

Should we put these in a comma separated list in an env variable?

File: packages/trpc/server/routers/viewer/organizations/update.handler.ts
Comment (zomars) [inline packages/trpc/server/routers/viewer/organizations/update.handler.ts:1]: Discarding changes to prevent conflicts with #9727

**Review by zomars — DISMISSED** [review]

**Comment by zomars** [comment]
- Moved constant to a env
- Added fallback when envs are undefined 
- Tested locally
- Fixed types

**Comment by zomars** [comment]
Envs already set on Vercel as well

**Comment by zomars** [comment]
Fixed unit tests. (my bad)

**Review by zomars — APPROVED** [review]

**Pull request merged**